### PR TITLE
Preliminary support for Go 1.22

### PR DESCRIPTION
**Pull request by eskultety**
This is just very basic support for 1.22 done by a simple version bump (note that the change comes without pre-installing a new version inside the container because that would require more code changes beyond just the Dockerfile).

# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] Docs updated (if applicable)
- [ ] Docs links in the code are still valid (if docs were updated)

**Note:** if the contribution is external (not from an organization member), the CI
pipeline will not run automatically. After verifying that the CI is safe to run:

- [approve GitHub Actions workflows][approve-gh-actions] by clicking a button
- approve the Red Hat Trusted App Pipeline container build by commenting `/ok-to-test`
  (as is the [standard for Pipelines as Code][pac-running-pipeline])

[approve-gh-actions]: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks
[pac-running-pipeline]: https://pipelinesascode.com/docs/guide/running/#running-the-pipelinerun

### CODE REVIEW

File: cachi2/core/package_managers/gomod.py [hunk 1]
```diff
@@ -1011,7 +1011,7 @@ def _get_go_work_path(app_dir: RootedPath) -> Optional[RootedPath]:
     go = Go()
     go_work_file = go(["env", "GOWORK"], {"cwd": app_dir}).rstrip()
 
-    if not go_work_file:
+    if not go_work_file or go_work_file == "off":
```
Comment (lkolacek) [inline cachi2/core/package_managers/gomod.py:1027]: Why we need disabling go_work_file?

**Comment by lkolacek** [comment]
LGTM, left 1 comment here for my education. Thanks.

File: cachi2/core/package_managers/gomod.py [hunk 2]
```diff
@@ -875,6 +882,12 @@ def _resolve_gomod(
         flags, app_dir, config.gomod_strict_vendor
     )
     if should_vendor:
+        if go_work_path and go_work_path.join_within_root("vendor").path.is_dir():
+            # NOTE: the same error will be reported even for 1.21 which doesn't support workspace
+            # vendoring, but given it's an invalid configuration and that we plan full 1.22 support
+            # in the foreseeable future, a not so user friendly error should be fine
+            raise UnsupportedFeature("Workspace vendoring introduced is not supported")
```
Comment (taylormadore) [inline cachi2/core/package_managers/gomod.py:889]: nit: Consider rephrasing the error message slightly. Perhaps something like "Go workspace vendoring is not supported"

**Review by taylormadore — APPROVED** [review]

File: cachi2/core/package_managers/gomod.py (hunk 1 above)
Comment (eskultety) [reply cachi2/core/package_managers/gomod.py:1027]: @lkolacek it's fixing a bug in our code base that could affect the check later down the road for this preliminary support.

**Review by brunoapimentel — APPROVED** [review]

**Pull request merged**